src/spikeinterface/core/tests/test_core_tools.py

```diff
@@ -163,32 +166,80 @@ def test_write_memory_recording():
 
 
-def test_recursive_path_modifier():
-    # this test nested depth 2 path modifier
-    d = {
-        "kwargs": {
-            "path": "/yep/path1",
-            "recording": {
-                "module": "mock_module",
-                "class": "mock_class",
-                "version": "1.2",
-                "annotations": {},
-                "kwargs": {"path": "/yep/path2"},
-            },
+def test_path_utils_functions():
+    if platform.system() != "Windows":
+        # posix path
+        d = {
+            "kwargs": {
+                "path": "/yep/sub/path1",
+                "recording": {
+                    "module": "mock_module",
+                    "class": "mock_class",
+                    "version": "1.2",
+                    "annotations": {},
+                    "kwargs": {"path": "/yep/sub/path2"},
+                },
+            }
         }
-    }
 
-    d2 = recursive_path_modifier(d, lambda p: p.replace("/yep", "/yop"))
-    assert d2["kwargs"]["path"].startswith("/yop")
-    assert d2["kwargs"]["recording"]["kwargs"]["path"].startswith("/yop")
+        d2 = recursive_path_modifier(d, lambda p: p.replace("/yep", "/yop"))
+        assert d2["kwargs"]["path"].startswith("/yop")
+        assert d2["kwargs"]["recording"]["kwargs"]["path"].startswith("/yop")
+
+        d3 = make_paths_relative(d, Path("/yep"))
+        assert d3["kwargs"]["path"] == "sub/path1"
+        assert d3["kwargs"]["recording"]["kwargs"]["path"]  == "sub/path2"
+
+        d4 = make_paths_absolute(d3, "/yop")
+        assert d4["kwargs"]["path"].startswith("/yop")
+        assert d4["kwargs"]["recording"]["kwargs"]["path"].startswith("/yop")
+
+
+    if platform.system() == "Windows":
+        # test for windows Path
+        d = {
+            "kwargs": {
+                "path": "c:\\yep\\sub\\path1",
+                "recording": {
+                    "module": "mock_module",
+                    "class": "mock_class",
+                    "version": "1.2",
+                    "annotations": {},
+                    "kwargs": {"path": "c:\\yep\\sub\\path2"},
+                },
+            }
+        }
+
+        d2 = make_paths_relative(d, "c:\\yep")
+        # the str be must unix like path even on windows for more portability
+        assert d2["kwargs"]["path"] == "sub/path1"
+        assert d2["kwargs"]["recording"]["kwargs"]["path"]  == "sub/path2"
+
+        # same drive
+        assert check_paths_relative(d, "c:\\yep")
+        # not the same drive
+        assert not check_paths_relative(d, "d:\\yep")
+
+        d = {
+            "kwargs": {
+                "path": "\\host\share\yep\\sub\\path1",
+            }
+        }
+        # UNC cannot be relative to d: drive
+        assert not check_paths_relative(d, "d:\\yep")
+
+        # UNC can be relative to the same UNC
+        assert check_paths_relative(d, "\\host\share")
```

Comment:
```suggestion
        assert check_paths_relative(d, r"\\host\share")
```

I typically advise and most people typically learn through their own reading that they should always just use raw strings for Windows. We probably want to make sure tests pass compared to the most common default Windows behavior (and then document the appropriate behavior for users). --one of the tests is warning about inappropriate escaping as well.

Reply:
Oh thanks! I fogot that one!